asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
Making Add idempotent for an identical, current assertion is the exact use case I have for this error type. :)

Moving the idempotent check into the asserts package is doable. It would obviously be a small change in behaviour, but I can't imagine anything would break. I think an idempotent check would be best handled by Database.Add so each backstore doesn't have to deal with the issue. We'd still need this error type, so I would suggest a new branch if we decide to go that way.

Two different assertions with the same revision is not necessarily as bad as you suggest. It could just be two concurrent processes trying to update an assertion ... just a conflict due to an optimistic lock style check really.